plugins/BEdita/Core/tests/TestCase/Model/Entity/DateRangeTest.php

```diff
@@ -0,0 +1,652 @@
+                ],
+            ],
+            [
+                true,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-02-01',
+                ],
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                true,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => '2017-02-10',
+                ],
+            ],
+            [
+                true,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-02-01',
+                    'end_date' => '2017-02-10',
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => '2017-01-10',
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-01-01',
+                    'end_date' => null,
+                ],
+            ],
+            [
+                false,
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+                [
+                    'start_date' => '2017-01-05',
+                    'end_date' => null,
+                ],
+            ],
+        ];
```

Comment:
sometimes input data can be bad; maybe some "wrong data" tests could be useful. For instance:
```
[ // wrong interval; end_date before start_date
    'start_date' => '2017-01-05',
    'end_date' => '2017-01-04'
],
[ // unhandled or wrong data format
    'start_date' => '01 2017 31', // odd format
    'end_date' => '2017 02 0' // truncated
],
[ // wrong characters for date
    'start_date' => 'p2*0170x0ab',
    'end_date' => '2017 01 20'
],
[ // wrong interval: start_date null not possible (or is it possible?)
    'start_date' => null,
    'end_date' => '2017 01 20'
],
[ // both start_date and end_date null (uh, what then?)
    'start_date' => null,
    'end_date' => null
],
```